apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuery.cs

```diff
@@ -66,17 +65,26 @@ internal async Task<AggregateQuerySnapshot> GetSnapshotAsync(ByteString transact
         IAsyncEnumerable<RunAggregationQueryResponse> responseStream = GetAggregationQueryResponseStreamAsync(transactionId, cancellationToken);
         Timestamp? readTime = null;
         long? count = null;
+        Dictionary<string, Value> data = new Dictionary<string, Value>();
         await responseStream.ForEachAsync(response => ProcessResponse(response), cancellationToken).ConfigureAwait(false);
         GaxPreconditions.CheckState(readTime != null, "The stream returned from RunAggregationQuery did not provide a read timestamp.");
-        return new AggregateQuerySnapshot(this, readTime.Value, count);
+        return new AggregateQuerySnapshot(this, readTime.Value, count, data);
```

Comment:
I'd prefer us not to collect the count in this code at all - let's just pass the dictionary to `AggregateQuerySnapshot`, and that can validate that it's an integer if it's present.